errors/errors.go

```diff
@@ -125,32 +126,67 @@ type Error struct {
 }
 
 // Error returns the stored description
-func (e Error) Error() string { return e.desc }
+func (e *Error) Error() string { return e.desc }
```

Comment:
Why make all usages *Error not Error?
Just so you get the `nil.Is()` comparison short-cut for noErr?

Reply:
I think it is dangerous when using a static method like this. Copy works only for the simple types. It could lead to unwanted behavior when the structure is containing a pointer of a slice.

Should I change it back to static methods?

I only need the pointer for the `Is` method. That makes tests easier to write. You can expect in a test a `nil` error and comparison will ensure it.

Reply:
Sure. But `Error` is an int and a string.

I am not strongly opposed to it, but it did raise my eyebrows. I would prefer to use "static" calls (copy of struct) when possible. Maybe just use `*Error` for Is?

But I don't want to be a blocker for no reason. So, I would like an opinon from @alpe If you both are happy with this change, then go ahead. If he has concerns as well, then we can change.

Reply:
I will change back to static method for all but `Is`.

Reply:
I was stumbling about the pointers, too. In terms of API design we should stick to the "value" methods we had before as any call to this and the other functions does not modify the internal state of an error object. In Go this is communicated by "value" methods vs "pointers" for mutable state.

Reply:
Fair enough. Good to merge in my mind